core/server/worker/src/main/java/alluxio/worker/netty/NettyExecutors.java

```diff
@@ -34,25 +35,25 @@
   public static final ExecutorService BLOCK_READER_EXECUTOR =
       new ThreadPoolExecutor(THREADS_MIN,
           Configuration.getInt(PropertyKey.WORKER_NETWORK_NETTY_BLOCK_READER_THREADS_MAX),
-          THREAD_STOP_MS, TimeUnit.MILLISECONDS, new SynchronousQueue<Runnable>(),
```

Comment:
Without this, the submit() call fails if all the threads in the pool are active which is not expected. Instead, the tasks should be pushed to the queue.  I think i can reduce the threadpool max size after this change (not changed for now since i did not test that).